ui/src/containers/CreateClient.js

```diff
@@ -1,217 +1,162 @@
+      resultDetails: null,
+      validated: false,
+      showModal: true, 
+      loading: false,
+      selectedPlatform: null
+    });
+  };
 
-    open = () => {
-      if (this.state.resetOnStart) { // resets the state if it was already there
-        this.setState({
-          clientConfiguration: null,
-          resultDetails: null,
-          activeStepIndex: WIZARD_SELECT_PLATFORM,
-          validated: false
-        });
-      }
-      this.setState({ showModal: true, loading: false });
-    };
+  selectPlatform = (state) => {
 
-    configureClient = (state) => {
-      const newState = { clientConfiguration: state.clientConfiguration };
-      newState.validated = state.validation && state.validation[CREATE_CLIENT_NAME] && state.validation[CREATE_CLIENT_APP_ID] && !Object.values(state.validation).find(v => v === 'error'); // check if everything was filled in and validated
-      this.setState(newState);
-    }
+    var newState = {...this.state, valid: false};
 
-    renderClientSpecificForm = () => {
-      switch (this.state.clientConfiguration[CREATE_CLIENT_TYPE]) {
-        case PLATFORM_ANDROID:
-          return <CreateAndroidClient configureClient={this.configureClient} clientConfiguration={this.state.clientConfiguration} />;
-        case PLATFORM_CORDOVA:
-          return <CreateCordovaClient configureClient={this.configureClient} clientConfiguration={this.state.clientConfiguration} />;
-        case PLATFORM_IOS:
-          return <CreateIOSClient configureClient={this.configureClient} clientConfiguration={this.state.clientConfiguration} />;
-        case PLATFORM_XAMARIN:
-          return <CreateXamarinClient configureClient={this.configureClient} clientConfiguration={this.state.clientConfiguration} />;
-        default:
-          return null;
-      }
+    switch (state.id) {
+      case 'platform_android': newState.selectedPlatform = PLATFORM_ANDROID; break;
+      case 'platform_ios': newState.selectedPlatform = PLATFORM_IOS; break;
+      case 'platform_cordova': newState.selectedPlatform = PLATFORM_CORDOVA; break;
+      case 'platform_xamarin': newState.selectedPlatform = PLATFORM_XAMARIN; break;
+      default: throw new Error(`unsupported platform ${state.id}`);
     }
+    this.setState(newState);
+  }
 
+  renderPlatformSelection() {
+    return (
+      <FormGroup>
+        <ControlLabel className="required">
+          Application Platform
+        </ControlLabel>
+        <PlatformItems itemSelected={this.selectPlatform} >
+          <PlatformItem title="Android" id="platform_android" inclass="fa fa-android"/>
+          <PlatformItem title="Cordova" id="platform_cordova">
+            <span><img src="../../img/cordova.png" alt="Cordova" /></span>
+          </PlatformItem>
+          <PlatformItem title="iOS" id="platform_ios" inclass="fa fa-apple" />
+          <PlatformItem title="Xamarin" id="platform_xamarin" >
+            <span><img src="../../img/xamarin.svg" alt="Xamarin" /></span>
```

Comment:
can we maybe move the logic of either using the `fa` icon or image to the `PlatformItem` itself, so that when it's been used by other components, all that needed to be provided is the name of the platform.